graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+from typing import Union, Tuple
+from .base import BaseVN
+from ..embed import BaseSpectralEmbed
+from ..embed import AdjacencySpectralEmbed as ase, LaplacianSpectralEmbed as lse
+import numpy as np
+from scipy.spatial import distance
+
+
+class SpectralVertexNomination(BaseVN):
+    """
+    Class for spectral vertex nomination on a single graph.
+
+    Given a graph :math:`G=(V,E)` and a subset of :math: `V` called :math: `S` (the "seed"), Single Graph Vertex
+    Nomination is the problem of ranking all V in order of relation to members of S. Spectral Vertex Nomination solves
+    this problem by embedding G into a low dimensional euclidean space (:ref:`tutorials <embed_tutorials>`), and then
+    generating a nomination list by some distance based algorithm. In the simple unattributed case, for each seed vertex
+    u, the other vertices are ranked in order of euclidean distance from u. In the attributed case, vertices are ranked
+    by relatedness to each attribute present in the set of seed vertices.
+
+    Parameters
+    ----------
+    embedding: np.ndarray, default = None
+        A pre-calculated embedding may be provided, in which case it will be used for vertex nomination instead of
+        embedding the adjacency matrix using `embedder`.
+    embedder: str or BaseEmbed, default = 'ASE'
+        May provide either a embed object or a string indicating which embedding method to use, which may be either:
+        "ASE" for :py:class:`~graspologic.embed.AdjacencySpectralEmbed` or
+        "LSE" for :py:class:`~graspologic.embed.LaplacianSpectralEmbed`.
+    persistent : bool, default = True
+        If ``False``, future calls to fit will overwrite an existing embedding. Must be ``True`` if an embedding is
+        provided.
+
+    Attributes
+    ----------
+    attr_labels_ : np.ndarray
+        The attributes of the vertices in the seed (parameter 'y' for fit).
+        Shape is the number of seed vertices. Each value is unique in the unattributed case.
+    unique_att_ : np.ndarray
```

Comment:
```suggestion
    unique_attributes_ : np.ndarray
```

Reply:
didn't think "att" or "attr" were immediately recognizable as abbreviations for attribute.